Mathlib/Data/Set/Basic.lean

```diff
@@ -1521,6 +1521,36 @@ theorem eq_empty_of_ssubset_singleton {s : Set α} {x : α} (hs : s ⊂ {x}) : s
   ssubset_singleton_iff.1 hs
 #align set.eq_empty_of_ssubset_singleton Set.eq_empty_of_ssubset_singleton
 
+theorem Nonempty.eq_singleton_of_nonempty [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {default} := h.subset_singleton_iff.mp (fun x _ => by simp [Unique.uniq _ x])
+
+theorem Nonempty.eq_singleton_zero_of_nonempty [Zero α] [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {0} := by convert h.eq_singleton_of_nonempty
+
+theorem Nonempty.eq_singleton_one_of_nonempty [One α] [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {1} := by convert h.eq_singleton_of_nonempty
+
+noncomputable def uniqueOfNonempty [Subsingleton α] {s : Set α} (h : s.Nonempty) :
+    Unique α := uniqueOfSubsingleton h.choose
+
+theorem Nonempty.eq_singleton_of_nonempty_of_subsingleton [Subsingleton α] {s : Set α}
```

Comment:
this seems like a footgun, for just `Subsingleton` I'd go with `s = {h.choose}`

Reply:
I am not sore about that part indeed. This is the code from @Ruben-VandeVelde. I don't feel confortable enough with this part of the library to know if it's the right way to do it.